services/slam/internal/testhelper/testhelper.go

```diff
@@ -23,3 +28,18 @@ type Service interface {
 	GetSLAMProcessConfig() pexec.ProcessConfig
 	GetSLAMProcessBufferedLogReader() bufio.Reader
 }
+
+// TestComparePointCloudsFromPCDs is a helper function for checking GetPointCloudMapFull response along with associated pcd validity checks.
+func TestComparePointCloudsFromPCDs(t *testing.T, pcdInput, pcdOutput []byte) {
+	pcInput, err := pointcloud.ReadPCD(bytes.NewReader(pcdInput))
+	test.That(t, err, test.ShouldBeNil)
+	pcOutput, err := pointcloud.ReadPCD(bytes.NewReader(pcdOutput))
+	test.That(t, err, test.ShouldBeNil)
+
+	pcInput.Iterate(0, 0, func(p r3.Vector, d pointcloud.Data) bool {
```

Comment:
We additionally need to check that the metadata (including number of points) is the same. Otherwise, this function would succeed even if pcdInput is a subset of pcdOutput

Reply:
We do, do a direct byte-to-byte comparison before this which should handle that (and technically this pointcloud check as well) but can add meta data checks if desirable as well.